doctr/models/recognition/master/pytorch.py

```diff
@@ -0,0 +1,290 @@
+    'master': {
+        'mean': (.5, .5, .5),
+        'std': (1., 1., 1.),
+        'input_shape': (3, 48, 160),
+        'post_processor': 'MASTERPostProcessor',
+        'vocab': VOCABS['french'],
+        'url': None,
+    },
+}
+
+
+class MAGC(nn.Module):
+
+    """Implements the Multi-Aspect Global Context Attention, as described in
+    <https://arxiv.org/pdf/1910.02562.pdf>`_.
+
+    Args:
+        inplanes: input channels
+        headers: number of headers to split channels
+        att_scale: if True, re-scale attention to counteract the variance distibutions
+        **kwargs
+    """
+
+    def __init__(
+        self,
+        inplanes: int,
+        headers: int = 1,
+        att_scale: bool = False,
+        **kwargs
+    ) -> None:
+        super().__init__(**kwargs)
+
+        self.headers = headers  # h
+        self.inplanes = inplanes  # C
+        self.att_scale = att_scale
+
+        self.single_header_inplanes = int(inplanes / headers)  # C / h
+
+        self.conv_mask = nn.Conv2d(self.single_header_inplanes, 1, kernel_size=1)
+        self.softmax = nn.Softmax(dim=2)
+
+        self.channel_add_conv = nn.Sequential(
+            nn.Conv2d(self.inplanes, self.inplanes, kernel_size=1),
+            nn.LayerNorm([self.inplanes, 1, 1]),
+            nn.ReLU(inplace=True),
+            nn.Conv2d(self.inplanes, self.inplanes, kernel_size=1)
+        )
+
+    def context_modeling(self, inputs: torch.Tensor) -> torch.Tensor:
+        batch, _, height, width = inputs.size()
+        # [N*headers, C', H , W] C = headers * C'
+        x = inputs.view(batch * self.headers, self.single_header_inplanes, height, width)
+        shortcut = x
+
+        # [N*headers, C', H * W] C = headers * C'
+        # input_x = input_x.view(batch, channel, height * width)
+        shortcut = shortcut.view(batch * self.headers, self.single_header_inplanes, height * width)
+
+        # [N*headers, 1, C', H * W]
+        shortcut = shortcut.unsqueeze(1)
+        # [N*headers, 1, H, W]
+        context_mask = self.conv_mask(x)
+        # [N*headers, 1, H * W]
+        context_mask = context_mask.view(batch * self.headers, 1, height * width)
+
+        # scale variance
+        if self.att_scale and self.headers > 1:
+            context_mask = context_mask / math.sqrt(self.single_header_inplanes)
+
+        # [N*headers, 1, H * W]
+        context_mask = self.softmax(context_mask)
+
+        # [N*headers, 1, H * W, 1]
+        context_mask = context_mask.unsqueeze(-1)
+        # [N*headers, 1, C', 1] = [N*headers, 1, C', H * W] * [N*headers, 1, H * W, 1]
+        context = torch.matmul(shortcut, context_mask)
+
+        # [N, headers * C', 1, 1]
+        context = context.view(batch, self.headers * self.single_header_inplanes, 1, 1)
+
+        return context
+
+    def forward(self, inputs: torch.Tensor, **kwargs) -> torch.Tensor:
+        # Context modeling: B, C, H, W  ->  B, C, 1, 1
+        context = self.context_modeling(inputs)
```

Comment:
Mmmh this is a tricky topic, but for clarity, I'd suggest writing the commands in forward. I might be wrong but here are my thoughts:
if I read through the forward, I can see two calls to methods/attributes. I'm mentally guessing those are modules you instantiated in the constructor, which isn't the case for context_modeling here. If we unroll this function, the reader can see that all `self....` calls are from modules (with parameters). Mixing both looks odd to me. What do you think?

Reply:
I see, I can change it. Do I need to change it in tensorflow as well ?